tests/blockchain/test_blockchain.py

```diff
@@ -3509,8 +3509,15 @@ async def test_reorg_new_ref(empty_blockchain, bt):
         fork_info: Optional[ForkInfo] = None
         if i < 10:
             expected = AddBlockResult.ALREADY_HAVE_BLOCK
-        elif i < 20:
+        elif i < 19:
             expected = AddBlockResult.ADDED_AS_ORPHAN
+        elif i == 19:
+            # same height as peak decide by iterations
+            peak = b.get_peak()
+            if block.total_iters < peak.total_iters:
+                expected = AddBlockResult.NEW_PEAK
+            else:
+                expected = AddBlockResult.ADDED_AS_ORPHAN
```

Comment:
Coverage says this code is not exercised, but I think we should. One way would be to parametrize the test with a boolean to decide in which case we're in (`NEW_PEAK` or `ORPHAN`), and depending on that, to change the order in which we add the chains (chain1-chain2 or chain2-chain1 in each test case, depending which chain has the heaviest 19th block).

Reply:
@fchirica please check if this review comment is still relevant, and resolve it if not. Open threads block merge.